core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -646,8 +649,12 @@ private[spark] class TaskSetManager(
             s"${ef.className} (${ef.description}) [duplicate $dupCount]")
         }
 
+      case e: TaskCommitDenied =>
+        logWarning(failureReason)
+
       case e: TaskFailedReason =>  // TaskResultLost, TaskKilled, and others
         logWarning(failureReason)
+        return
```

Comment:
Ah, this `return` is probably why we're failing tests, since it causes us to completely bypass the code for counting the number of task failures.